orchestrator/orchestrator.go

```diff
@@ -119,63 +130,82 @@ func (s *Orchestrator) execute(filter func(wf *process.Process, node *process.Pr
 	var processes []*process.Process
 	route := fmt.Sprintf("custom/%s/%s", processmodule.QuerierRoute, processmodule.QueryList)
 	if err := s.rpc.QueryJSON(route, nil, &processes); err != nil {
-		s.ErrC <- err
+		s.logger.Error(err.Error())
 		return
 	}
 	for _, wf := range processes {
 		for _, node := range s.findNodes(wf, filter) {
 			if err := s.executeNode(wf, node, exec, event, data); err != nil {
-				s.ErrC <- err
+				s.logger.With(keyvals(wf, node, exec, event, data)...).Error(err.Error())
 			}
 		}
 	}
 }
 
 func (s *Orchestrator) executeNode(wf *process.Process, n *process.Process_Node, exec *execution.Execution, event *event.Event, data *types.Struct) error {
-	logrus.WithField("module", "orchestrator").
-		WithField("node.key", n.Key).
-		WithField("type", fmt.Sprintf("%T", n)).Debug("process process")
-	if task := n.GetTask(); task != nil {
+	log := s.logger.With(keyvals(wf, n, exec, event, data)...)
+	// Process the node
+	switch x := n.Type.(type) {
+	case *process.Process_Node_Task_:
 		// This returns directly because a task cannot process its children.
 		// Children will be processed only when the execution is done and the dependencies are resolved
-		return s.processTask(n.Key, task, wf, exec, event, data)
-	} else if m := n.GetMap(); m != nil {
+		createdExecHash, err := s.processTask(n, x.Task, wf, exec, event, data)
+		if err != nil {
+			return err
+		}
+		log.With("createdExecHash", createdExecHash.String()).Info("execution created")
+		return nil // stop workflow execution
+	case *process.Process_Node_Map_:
 		var err error
-		data, err = s.processMap(n.Key, m.Outputs, wf, exec, data)
+		data, err = s.processMap(n.Key, x.Map.Outputs, wf, exec, data)
 		if err != nil {
 			return err
 		}
-	} else if filter := n.GetFilter(); filter != nil {
-		if !s.filterMatch(filter, wf, n.Key, exec, data) {
-			return nil
+		if result, err := json.Marshal(data); err == nil {
+			log = log.With("output", string(result))
+		}
+	case *process.Process_Node_Filter_:
+		if result, err := json.Marshal(x.Filter); err == nil {
+			log = log.With("filter", string(result))
+		}
+		if !s.filterMatch(x.Filter, wf, n, exec, data) {
+			log.Info("filter does not match data")
+			return nil // stop workflow execution
 		}
```

Comment:
Should we add a log when the filter matches as well?
```suggestion
		if s.filterMatch(x.Filter, wf, n, exec, data) {
			log.Info("filter matches data")
	        } else {
	 		log.Info("filter does not match data")
			return nil // stop workflow execution
		}
```

Reply:
When the filter matches, the execution continue (no return here) and the log is displayed like any others at line 186:
```
log.Info("executed process transition")
```